packages/check/src/main.rs

```diff
@@ -86,17 +95,32 @@ pub fn main() {
 }
 
 fn check_contract(
-    path: impl AsRef<Path>,
+    path: &str,
     available_capabilities: &HashSet<String>,
+    verbose: bool,
 ) -> anyhow::Result<()> {
     let mut file = File::open(path)?;
 
     // Read wasm
     let mut wasm = Vec::<u8>::new();
     file.read_to_end(&mut wasm)?;
 
+    // Potentially lossy filename or path as used as a short prefix for the output
+    let filename_identifier: String = Path::new(path)
+        .file_name()
+        .map(|f| String::from_utf8_lossy(f.as_encoded_bytes()).to_string())
```

Comment:
I think this would be a bit cleaner:
```suggestion
        .map(|f| f.to_string_lossy().into_owned())
```

Reply:
Oh nice. Did you test this diff?

Reply:
Anyway, let the CI try it out